### PR TITLE
Rejoin shared instance

### PR DESCRIPTION
**Pull request by AlexVild**
Allows ex-party members to rejoin a shared instance

### CODE REVIEW

File: Intersect.Server.Core/Entities/Player.cs [hunk 1]
```diff
@@ -4906,42 +4940,42 @@ public void AddParty(Player target)
...
-            else
+
+            target.LeaveParty();
+            Party.Add(target);
+
+            //Update all members of the party with the new list
+            for (var i = 0; i < Party.Count; i++)
             {
-                PacketSender.SendChatMsg(this, Strings.Parties.limitreached, ChatMessageType.Party, CustomColors.Alerts.Error);
+                Party[i].Party = Party;
```
Comment (lodicolo) [inline Intersect.Server.Core/Entities/Player.cs:4972]: Cache Party[i]

File: Intersect.Server.Core/Entities/Player.cs [hunk 2]
```diff
@@ -2125,11 +2125,45 @@ public bool CanChangeToInstanceType(MapInstanceType? instanceType, bool fromLogi
...
-                        isValid = false;
+                        // Check to see if the instance still exists & is populated
+                        if (!InstanceProcessor.TryGetInstanceController(SharedMapInstanceId, out var sharedController) || sharedController.Players.Count == 0)
+                        {
+                            return false;
+                        }
+
+                        // If it does, get the party leader from it...
+                        var partyMember = sharedController.Players.FirstOrDefault();
+                        if (partyMember == default)
```
Comment (lodicolo) [inline Intersect.Server.Core/Entities/Player.cs:2136]: What happens if _you_ were the party leader? I actually have no idea what happens to a 3+ person party if the lead disconnects

**Review by lodicolo — CHANGES_REQUESTED** [review]

File: Intersect.Server.Core/Entities/Player.cs (hunk 2 above)
Comment (AlexVild) [reply Intersect.Server.Core/Entities/Player.cs:2136]: The leadership goes to the next in line of the array.

File: Intersect.Server.Core/Entities/Player.cs (hunk 1 above)
Comment (AlexVild) [reply Intersect.Server.Core/Entities/Player.cs]: Done

**Review by lodicolo — APPROVED** [review]

**Pull request merged**